### PR TITLE
Connect api

### PR DESCRIPTION
**Pull request by Zuhaib042**
### Crazy-Books milestone-04
**Summary**
- I used fetch to make calls to Bookstore API.
- the app fetches data from API on load.
- Refactor action creators to store data in API.

### CODE REVIEW

**Review by BertrandConxy — COMMENTED** [review]
Hi @Zuhaib042, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no README file. Please add a README file

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/books/books.js [hunk 1]
```diff
@@ -35,20 +18,38 @@ const addremoveReducer = (state = data, action = {}) => {
...
-  type: ADD_BOOK,
-  payload,
-});
-
-export const removeBook = (payload) => ({
-  type: REMOVE_BOOK,
-  payload,
-});
+
+export const fetchBooks = () => async (dispatch) => {
```
Comment (addod19) [inline src/redux/books/books.js]: - [ ] According to the project requirements, we are required to use `createAsyncThunk` for creating our thunk action creators. Correct me if I'm wrong, I do not see that you have used it
Screenshot from 2022-11-10 18-03-54

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @Zuhaib042 and @BertrandConxy, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Add and remove book functionality works as expected :rocket: 
- [ ] Action creator for removing a book is refactored :+1: 
- [ ] Action creator for adding a book is refactored :100: 
- [ ] Bookstore API call is done with `fetch` method :fireworks: 
 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (Zuhaib042) [reply src/redux/books/books.js]: I think the project requirement is to either use `fetch` or `axios` with `createAsyncThunk`.
so I am not using axios and not using createAsyncThunk.

Comment (Zuhaib042) [reply src/redux/books/books.js]: I believe I am fulfilling the project requirement and some of my coding partners also didn't use createAsyncThunk with Axios and their projects got approved.

Comment (Ismailco) [inline src/redux/books/books.js]: - [ ] As the previous reviewer mentioned kindly use `createAsyncThunk` for creating thunk action creators, as per the requirement you should use `fetch` or `Axios` either way you need to use `createAsyncThunk`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Zuhaib042, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ ✔️ ✔️

Hi @Zuhaib042, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Good commit history ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**